src/billing/billing-address-action-creator.ts

```diff
@@ -15,14 +16,42 @@ export default class BillingAddressActionCreator {
         private _checkoutClient: CheckoutClient
     ) {}
 
-    updateAddress(
-        address: Partial<BillingAddressUpdateRequestBody>,
+    continueAsGuest(
+        credentials: GuestCredentials,
         options?: RequestOptions
     ): ThunkAction<UpdateBillingAddressAction, InternalCheckoutSelectors> {
         return store => Observable.create((observer: Observer<UpdateBillingAddressAction>) => {
+            const state = store.getState();
+            const checkout = state.checkout.getCheckout();
+
+            if (!checkout) {
+                throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+            }
+
+            const customer = state.customer.getCustomer();
+
+            if (customer && !customer.isGuest) {
+                throw new StandardError('Cannot continue as guest: customer is logged in.');
+            }
+
+            const billingAddress = state.billingAddress.getBillingAddress();
+
+            let billingAddressRequestBody;
+
+            if (!billingAddress) {
+                billingAddressRequestBody = credentials;
+            } else {
+                const { country, ...existingBillingAddressRequestBody } = billingAddress;
```

Comment:
A question - why do we need to omit the country field?

Reply:
because we send the `countryCode`, `country` is not part of the payload

Reply:
I see, does it break if we don't omit it?

Reply:
yeah, linting error, we have well defined types for request vs response